src/vcf/meta_entry.cpp

```diff
@@ -216,16 +219,28 @@ namespace ebi
             type_field != "String") {
             throw new MetaSectionError{entry.line, "INFO metadata Type is not a Integer, Float, Flag, Character or String"};
         }
+        
+        std::map<std::string, std::pair<std::string, std::string>> version_info_type_map;
+        std::map<std::string, std::pair<std::string, std::string>> version_info_number_map;
 
-        for (const auto & t : info_type) {
+        if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
+            version_info_type_map = info_type_v41_v42;
+            version_info_number_map = info_number_v41_v42;
+        }
+        else if (entry.source->version == Version::v43) {
+            version_info_type_map = info_type_v43;
+            version_info_number_map = info_number_v43;
+        }
```

Comment:
I don't like much that map copy. In this case I think it's better to extract the two `for`s to a function, and here do the call with the maps of one version or the other:
```
if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
    check_predefined_type_and_number(info_type_v41_v42, info_number_v41_v42);
} else {
    check_predefined_tags_type_and_number(info_type_v43, info_number_v43);
}
```